packages/connection-form/src/hooks/use-connect-form.ts

```diff
@@ -289,13 +312,55 @@ function handleUpdateHost({
   }
 }
 
+function isConnectionUpdatePersonalizationAction(
+  action: ConnectionFormFieldActions
+): action is UpdateConnectionPersonalizationAction {
+  return action.type === 'update-connection-personalization';
+}
+
+export function handleConnectionFormUpdateForPersonalization(
+  action: ConnectionFormFieldActions,
+  personalization: ConnectionPersonalizationOptions,
+  connectionString?: string
+): ConnectionPersonalizationOptions {
+  if (!isConnectionUpdatePersonalizationAction(action)) {
+    if (!personalization.isNameDirty && connectionString) {
+      try {
+        const parsedConnString = new ConnectionString(connectionString);
+        const name = parsedConnString.hosts.join(',');
+        return {
+          ...personalization,
+          name,
+        };
+      } catch (ex) {
+        // just keep previous value
+      }
+    }
+
+    return personalization;
+  }
+
+  const isNameDirty = action.isNameDirty || personalization.isNameDirty;
+  const name = action.name || personalization.name;
```

Comment:
There is a bug where I can't remove the last letter of the name after entering it due to this usage of OR operator and how it behaves with falsy values. Is this intended or should this fallback to current state value only in case of the action value being null / undefined?